presto-sqlserver/src/main/java/io/prestosql/plugin/sqlserver/SqlServerClient.java

```diff
@@ -387,4 +420,27 @@ public void setNull(PreparedStatement statement, int index)
             }
         };
     }
+
+    private static class DataCompressionDao
+    {
+        private final Handle handle;
+
+        private DataCompressionDao(Handle handle)
+        {
+            this.handle = requireNonNull(handle, "handle is null");
+        }
+
+        Optional<DataCompression> getDataCompression(JdbcTableHandle table)
+        {
+            return handle.createQuery("" +
+                    "SELECT data_compression_desc FROM sys.partitions p INNER JOIN sys.tables t ON p.object_id = t.object_id " +
+                    "WHERE SCHEMA_NAME(t.schema_id) = :schema AND t.name = :table_name " +
+                    "AND p.partition_number = 1")
+                    .bind("schema", table.getSchemaName())
+                    .bind("table_name", table.getTableName())
+                    .mapTo(String.class)
+                    .findOne()
+                    .flatMap(dataCompression -> Enums.getIfPresent(DataCompression.class, dataCompression).toJavaUtil());
```

Comment:
does `getIfPresent` supports `-` replacement with `_` or does it compare it case insensitively?  No change request.